client/errors.go

```diff
@@ -142,6 +142,27 @@ const (
 
 	// ErrorKindValidationSetNotFound: validation set cannot be found.
 	ErrorKindValidationSetNotFound ErrorKind = "validation-set-not-found"
+
+	// ErrorKindAppArmorPromptingNotRunning: AppArmor Prompting is not running.
+	ErrorKindAppArmorPromptingNotRunning ErrorKind = "apparmor-prompting-not-running"
+
+	// ErrorKindInterfacesRequestsPromptNotFound: interfaces requests prompt not found.
+	ErrorKindInterfacesRequestsPromptNotFound ErrorKind = "interfaces-requests-prompt-not-found"
+
+	// ErrorKindInterfacesRequestsRuleNotFound: interfaces requests rule not found.
+	ErrorKindInterfacesRequestsRuleNotFound ErrorKind = "interfaces-requests-rule-not-found"
+
+	// ErrorKindInterfacesRequestsInvalidFields: POST body to prompting API contains invalid fields.
+	ErrorKindInterfacesRequestsInvalidFields ErrorKind = "interfaces-requests-invalid-fields"
+
+	// ErrorKindInterfacesRequestsReplyNotMatchRequestedPath: the path pattern in the prompt reply does not match the requested path.
+	ErrorKindInterfacesRequestsReplyNotMatchRequestedPath ErrorKind = "interfaces-requests-reply-not-match-requested-path"
+
+	// ErrorKindInterfacesRequestsReplyNotMatchRequestedPermissions: the permissions in the prompt reply do not match the requested permissions.
+	ErrorKindInterfacesRequestsReplyNotMatchRequestedPermissions ErrorKind = "interfaces-requests-reply-not-match-requested-permissions"
```

Comment:
Sounds good, but then it seems like this should be backed by an `invalidFieldValue` (just when marshalling) so that we can specify which (or both) field doesn't match.

Reply:
yes, the error value will need to be a map of field name to {Received,Request}

Reply:
I also wonder if Replied would make more sense here than Received ?